promdump/promdump.go

```diff
@@ -651,9 +770,52 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		var filename string
+		filename = *tarFilename
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			filename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(filename); err == nil {
+				log.Fatalf("Error: %s already exists. Please choose a different filename.\n", filename)
+				return
+			}
+		}
+
+		out, err := os.Create(filename)
+		if err != nil {
+			log.Printf("Error writing archive: %v (File: %s)\n", err, filename)
+			return
+		}
+
+		// Create the archive
+		err = createArchive(out)
+		if err != nil {
+			log.Printf("Error creating archive: %v\n", err)
+		}
+		// cleaning files
+		for _, v := range collectMetrics {
+			metricName, err := getMetricName(v)
+			if err != nil {
+				log.Fatalf("Error removing files: %v", err)
+			}
+			if v.collect {
+				_, err := cleanFiles(metricName, v.fileCount)
+				if err != nil {
+					log.Fatalf("Error removing files: %v", err)
+				}
+			}
+		}
```

Comment:
How extensively have you tested this? Add debug messages showing which `v` it's actually working on in each iteration and I think you'll find it's not always doing the right thing.

This `for _, v := range collectMetrics` pattern creates what's called a [closure](https://en.wikipedia.org/wiki/Closure_(computer_programming)). You probably need to change this to the following to "break" that closure:
```
		for _, v := range collectMetrics {
			v := v
			metricName, err := getMetricName(v)
...
```

Reply:
used closure  as suggested 
```
for _, v := range collectMetrics {
			v := v
			metricName, err := getMetricName(v)
```